### DISCUSSION TITLE
How do I connect to WalletConnect from another Chrome Extension?

### DISCUSSION THREAD

**Discussion by javaspeak** [WalletConnect/walletconnect-monorepo#3602] [Developer Support] [answered]
In my chrome extension I open an app in a new tab so it looks like:

     chrome-extension://oahhaapnfpkfnbgeopliehfgenifngde/register.html

Is it possible to connect to a browser wallet like metamask via Web3Modal there?

When I try it cannot see that Meta Mask is available and when I print window.ethereum it is undefined.

However if I run register.html in the web browser outside the chrome extension window.ethereum is defined and Web3Modal sees MetaMask is there.

**Comment by javaspeak** [comment]
Note that Meta Mask does provide a port provider so Wallet Connect could try and connect via that if it does not find window.ethereum.

Why does Wallet Connect not provide that port functionality for at least Meta Mask?

That would allow chrome extensions wanting to connect to Meta Mask web browser extension possible.

I think other browser wallet providers do not currently support port providers.  However if every they did Wallet Connect could integrate with them.

**Comment by glitch-txs** [comment, marked as answer]
window.ethereum is injected by wallets itself in the web page
